tests/parser/ReservoirCouplingTests.cpp

```diff
@@ -30,23 +33,84 @@
 #include <opm/input/eclipse/EclipseState/Tables/TableManager.hpp>
 #include <opm/input/eclipse/Python/Python.hpp>
 #include <opm/common/utility/OpmInputError.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/OpmLog/StreamLog.hpp>
+
+#include <sstream>
+#include <string>
 
 using namespace Opm;
 namespace {
-Schedule make_schedule(const std::string& schedule_string) {
+Schedule make_schedule(const std::string& schedule_string, bool slave_mode) {
     Parser parser;
     auto python = std::make_shared<Python>();
     Deck deck = parser.parseString(schedule_string);
     EclipseGrid grid(10,10,10);
     TableManager table ( deck );
     FieldPropsManager fp( deck, Phases{true, true, true}, grid, table);
     Runspec runspec (deck);
-    Schedule schedule(deck, grid , fp, runspec, python);
+    Schedule schedule(deck, grid , fp, runspec, python, slave_mode);
     return schedule;
 }
+
+void addStringLogger(std::ostringstream& stream_buffer) {
+    auto string_logger = std::make_shared<StreamLog>(stream_buffer, Log::DefaultMessageTypes);
+    OpmLog::addBackend("MYLOGGER", string_logger);
+}
+
+void assertRaisesInputErrorException(const std::string& schedule_string, bool slave_mode, const std::string& exception_string) {
+    bool exception_thrown = false;
+    try {
+        make_schedule(schedule_string, slave_mode);
+    }
+    catch (const Opm::OpmInputError& e) {
+        BOOST_CHECK_EQUAL(std::string(e.what()), exception_string);
+        exception_thrown = true;
+    }
+    catch (const std::exception& e) {
+        BOOST_FAIL("Expected Opm::OpmInputError but caught std::exception");
+    }
+    catch (...) {
+        BOOST_FAIL("Expected Opm::OpmInputError but caught unknown exception");
+    }
+    BOOST_CHECK_MESSAGE(exception_thrown, "Expected Opm::OpmInputError not thrown");
```

Comment:
If we can call make_schedule twice, we could simplify this to:
```
BOOST_CHECK_TROW(make_schedule(schedule_string, slave_mode), Opm::OpmInputError);
try {
        make_schedule(schedule_string, slave_mode);
}
catch (const Opm::OpmInputError& e) {
        BOOST_CHECK_EQUAL(std::string(e.what()), exception_string);
}
```

Reply:
Yes I like this simplification.